app/controllers/translation_controller.rb

```diff
@@ -74,15 +74,15 @@ def get_language_and_authorize_user
   end
 
   def validate_language_and_user(locale, lang)
-    if !lang
-      raise(:edit_translations_bad_locale.t(locale: locale))
-    elsif !@user
-      raise(:edit_translations_login_required.t)
-    elsif !@user.is_successful_contributor?
+    raise(:edit_translations_bad_locale.t(locale: locale)) unless lang
+    raise(:edit_translations_login_required.t) unless @user
+
+    unless @user.is_successful_contributor?
       raise(:unsuccessful_contributor_warning.t)
-    elsif lang.official && !reviewer?
-      raise(:edit_translations_reviewer_required.t)
     end
+    return unless lang.official && !reviewer?
+
+    raise(:edit_translations_reviewer_required.t)
```

Comment:
Is this right? 
(I actually don't understand what `raise` does. Does it interrupt execution and return to one level up?)

Reply:
Raise interrupts all the way to the top and exits the program, unless it is caught somewhere above.  That's what rescue does: it catches some or all raised exceptions.  I don't remember where these exceptions are being caught off the top of my head.

Reply:
Thanks @pellaea. The changes should work fine.

Reply:
I wonder if the following would be easier to read?  I like that it ensures all of the criteria do the same thing: they are all about cases which raise exceptions.

  def validate_language_and_user(locale, lang)
    raise(:edit_translations_bad_locale.t(locale: locale)) unless lang
    raise(:edit_translations_login_required.t) unless @user
    raise(:unsuccessful_contributor_warning.t) \
      unless @user.is_successful_contributor?
    raise(:edit_translations_reviewer_required.t) \
      if lang.official && !reviewer?
  end

Reply:
I agree it would be easier to read if each line started consistently. 
I did it this way because of line-length issues. 
I'll see if continuing the line solves the problem.